lib/checker/checkercomponent.cpp

```diff
@@ -222,7 +223,8 @@ void CheckerComponent::CheckThreadProc()
 
 		Checkable::IncreasePendingChecks();
 
-		Utility::QueueAsyncCallback(std::bind(&CheckerComponent::ExecuteCheckHelper, CheckerComponent::Ptr(this), checkable));
```

Comment:
* [ ] (At least) Here you've effectively replaced the intrusive pointer with a plain `this`. Please see [here](https://github.com/Icinga/icinga2/blob/f3129627149d8b3e24ba33894f28e1c4971ae723/lib/remote/jsonrpcconnection.cpp#L46) for a crash-safe solution.